.github/workflows/run_test_suite.yaml

```diff
@@ -1,35 +1,66 @@
-on: [push, pull_request]
+name: Tests and Checks
 
-name: Run test suite
+on:
+  push:
+    branches: [ main ]
+
+  pull_request:
+    branches: [ main ]
 
 jobs:
-  run-test-suite:
+  run-tests-checks:
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        rust-toolchain:
+          - stable
+          - nightly
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/cache@v2
-        if: always()
-        with:
-          path: |
-            ~/.cargo
-            ./target
-          key: ${{ runner.os }}-cargo-artifacts-${{ hashFiles('**/Cargo.toml') }}
-      - name: 'Install environment packages'
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Cache Project
+        uses: Swatinem/rust-cache@v2
```

Comment:
Can you provide any color commentary about the qualities and advantages of this action?

Reply:
Will add a comment and link. Essentially, makes builds faster by being smarter about caching vs the reg gh-cache, also takes care of a ton of boilerplate.